internal/store/postgres/asset_repository.go

```diff
@@ -164,7 +164,11 @@ func (r *AssetRepository) GetByID(ctx context.Context, id string) (asset.Asset,
 }
 
 func (r *AssetRepository) GetByURN(ctx context.Context, urn string) (asset.Asset, error) {
-	ast, err := r.getWithPredicate(ctx, sq.Eq{"a.urn": urn})
+	return r.GetByURNWithTx(ctx, nil, urn)
```

Comment:
can we just use `getWithPredicateWithTx` here?

Reply:
We can, but it will redundant with GetByURNWithTx for error handling since the different only `tx = nil`. So, I think it would be better to use the GetByURNWithTx instead

Reply:
I changed my mind, using different error response so use getWithPredicateWithTx instead, thanks mas